pheweb/serve/data_access/db.py

```diff
@@ -333,27 +458,30 @@ class TabixGnomadDao(GnomadDB):
 
     def __init__(self, matrix_path):
         self.matrix_path = matrix_path
+        self.tabix_file =pysam.TabixFile(self.matrix_path, parser=None)
+        self.tabix_handles = defaultdict( lambda: pysam.TabixFile(self.matrix_path, parser=None))
+        self.headers = self.tabix_file.header[0].split('\t')
 
-    def get_variant_annotations(self, id_list):
-        split = [id.split(':') for id in id_list]
-        variants = [{'chrom': s[0].replace('chr', ''), 'pos': int(s[1]), 'ref': s[2], 'alt': s[3]} for s in split]
+    def get_variant_annotations(self, var_list):
         annotations = []
         t = time.time()
-        with pysam.TabixFile(self.matrix_path, parser=None) as tabix_file:
-            #headers = [s.lower() for s in tabix_file.header[0].split('\t')]
-            headers = tabix_file.header[0].split('\t')
-            for var_i, variant in enumerate(variants):
-                tabix_iter = tabix_file.fetch(variant['chrom'], variant['pos']-1, variant['pos'], parser=None)
-                for row in tabix_iter:
-                    split = row.split('\t')
-                    if split[3] == variant['ref'] and split[4] == variant['alt']:
-                        for i, s in enumerate(split):
-                            if (headers[i].startswith('AF') and split[i] != 'NaN'):
-                                split[i] = float(s)
-                        annotations.append({'id': id_list[var_i], 'var_data': {headers[i]: split[i] for i in range(0,len(split))}})
-                    else:
-                        #print(split[3] + ' - ' + variant['ref'] + ' / ' + split[4] + ' - ' + variant['alt'])
-                        pass
+        ##print("There are {} active tabix handles for gnomad".format( len(self.tabix_handles)))
+        for var_i, variant in enumerate(var_list):
+            #print("There are {} active tabix handles for gnomad. Current pid {}".format( len(self.tabix_handles), os.getpid()))
+
+            ### TODO get rid of this chr shit once the annotation files have been fixed
+            fetch_chr = str(variant.chr).replace("23","X").replace("24","Y").replace("25","Y")
+            tabix_iter = self.tabix_handles[ threading.get_ident() ].fetch(fetch_chr, variant.pos-1, variant.pos)
+            for row in tabix_iter:
+                split = row.split('\t')
+                if split[3] == variant.ref and split[4] == variant.alt:
+                    for i, s in enumerate(split):
+                        if (self.headers[i].startswith('AF') and split[i] != 'NaN'):
```

Comment:
myea we should really have datatypes for this annotation as well. For some reason the AF was hardcoded here....    
There are very limited number of annotations and it's less configuration if just reading all annotations and returning them. If such feature is needed it can be added then.